src/sqlancer/oceanbase/ast/OceanBaseConstant.java

```diff
@@ -0,0 +1,628 @@
+            } else if (rightVal instanceof OceanBaseIntConstant){
+                return OceanBaseConstant.createBoolean(val < rightVal.getInt());
+            } else if (rightVal instanceof OceanBaseDoubleConstant){
+                return OceanBaseConstant.createBoolean(val < rightVal.getDouble());
+            } else if (rightVal instanceof OceanBaseTextConstant){
+                return isLessThan(rightVal.castAsDouble());
+            } else{
+                throw new AssertionError(rightVal);
+            } 
+        }
+
+        @Override
+        public OceanBaseConstant isEquals(OceanBaseConstant rightVal) {
+            if (rightVal.isNull()) {
+                return OceanBaseConstant.createNullConstant();
+            } else if (rightVal instanceof OceanBaseIntConstant){
+                return OceanBaseConstant.createBoolean(val == rightVal.getInt());
+            } else if (rightVal instanceof OceanBaseDoubleConstant){
+                return OceanBaseConstant.createBoolean(val == rightVal.getDouble());
+            } else if (rightVal instanceof OceanBaseTextConstant){
+                return isEquals(rightVal.castAsDouble());
+            } else {
+                throw new AssertionError(rightVal);
+            }
+        }
+        @Override
+        public OceanBaseDataType getType() {
+            return OceanBaseDataType.DOUBLE;
+        }
+
+    }
+
+    public static class OceanBaseTextConstant extends OceanBaseConstant {
+
+        private final String value;
+        private final boolean singleQuotes;
+
+        public OceanBaseTextConstant(String value) {
+            this.value = value;
+            singleQuotes = Randomly.getBoolean();
+
+        }
+
+        private void checkIfSmallFloatingPointText() {
+            boolean isSmallFloatingPointText = isString() && asBooleanNotNull()
+                    && castAs(CastType.SIGNED).getInt() == 0;
+            if (isSmallFloatingPointText) {
+                throw new IgnoreMeException();
+            }
+        }
+        @Override
+        public boolean isNull() {
+            if(value.equalsIgnoreCase("NULL"))
+                return true;
+            else
+                return false;
+        }
+
+        @Override
+        public boolean isEmpty(){
+            //"" "    " 
+            if (value.length() ==0 ){
+                return true;
+            }else{
+                for(int i =0;i< value.length(); i++){
+                   String sub =value.substring(i, i+1);
+                   if(!sub.equals(" "))
+                      return false;
+                }
+                return true;
+            }    
+        }
+
+        @Override
+        public boolean asBooleanNotNull() {
+            for (int i = value.length(); i >= 1; i--) {
+                try {
+                    char currentChar = value.charAt(i-1);
+                    int currentVal= Integer.valueOf(currentChar);
+                    if (currentVal < 48||currentVal > 57)
```

Comment:
Having named constants for 48 and 57 would be useful (but I assume that one of the style checks will complain about this anyway).